app/Commands/Extra/NewCompile.hs

```diff
@@ -27,6 +27,20 @@ getOutputFile ext inp = \case
     let baseOutputFile = invokeDir <//> filename mainFile
     return (replaceExtension' (fileExtToString ext) baseOutputFile)
 
+getOutputDir ::
+  (Members '[App] r) =>
+  FileExt ->
+  Maybe (AppPath File) ->
+  Maybe (AppPath Dir) ->
+  Sem r (Path Abs Dir)
+getOutputDir ext inp = \case
+  Just out -> fromAppPathDir out
+  Nothing -> do
+    mainFile <- getMainFile inp
+    invokeDir <- askInvokeDir
+    let baseOutputDir = invokeDir <//> filename (replaceExtension' (fileExtToString ext) mainFile)
+    return $ absDir $ toFilePath baseOutputDir
```

Comment:
having some helper function like `pathFileToPathDir : Path b File -> path b Dir` would make this more readable. The helper function can go to `Juvix.Prelude.Path`.